dali/test/python/test_operator_numba_func.py

```diff
@@ -0,0 +1,127 @@
+import numpy as np
+import os
+from numba import cfunc, types, carray
+
+from nvidia.dali import pipeline_def
+import nvidia.dali as dali
+import nvidia.dali.fn as fn
+from test_utils import get_dali_extra_path
+
+test_data_root = get_dali_extra_path()
+lmdb_folder = os.path.join(test_data_root, 'db', 'lmdb')
+
+fun1_sig = types.void(types.CPointer(types.uint8),  # Pointer to output sample
+                    types.CPointer(types.int64),    # Pointer to output sample shape
+                    types.CPointer(types.uint8),    # Pointer to input sample
+                    types.CPointer(types.int64),    # Pointer to input sample shape
+                    types.int32)                    # number of sample dimensions
+@cfunc(fun1_sig, nopython=True)
+def set_all_values_to_255(out_ptr, out_shape_ptr, in_ptr, in_shape_ptr, ndim):
+    out_shape = carray(out_shape_ptr, ndim)
+    out_arr = carray(out_ptr, (out_shape[0], out_shape[1], out_shape[2]))
+    out_arr[:] = 255
+
+fun2_sig = types.void(types.CPointer(types.float32),    # Pointer to output sample
+                    types.CPointer(types.int64),        # Pointer to output sample shape
+                    types.CPointer(types.float32),      # Pointer to input sample
+                    types.CPointer(types.int64),        # Pointer to input sample shape
+                    types.int32)                        # number of sample dimensions
+@cfunc(fun2_sig, nopython=True)
+def set_all_values_to_float(out_ptr, out_shape_ptr, in_ptr, in_shape_ptr, ndim):
+    out_shape = carray(out_shape_ptr, ndim)
+    out_arr = carray(out_ptr, (out_shape[0], out_shape[1], out_shape[2]))
+    out_arr[:] = 0.5
+
+setup1_sig = types.void(
+                types.CPointer(types.int64),    # Output batch shape pointer
+                types.int32,                    # Number of dimensions in output shape
+                types.CPointer(types.int32),    # Output dtype pointer
+                types.CPointer(types.int64),    # Input batch shape pointer
+                types.int32,                    # Number of dimensions in input shape
+                types.int32,                    # Input dtype
+                types.int32,                    # Number of samples in the batch
+                types.int32,                    # Number of outputs
+                types.int32)                    # Number of inputs
+@cfunc(setup1_sig, nopython=True)
+def setup_fn1(out_shape_ptr, out1_ndim, out_dtype, in_shape_ptr, in1_ndim, in_dtype, num_samples, num_outputs, num_inputs):
+    in_arr = carray(in_shape_ptr, num_samples * out1_ndim)
+    out_arr = carray(out_shape_ptr, num_samples * in1_ndim)
+    out_arr[0] = in_arr[1]
+    out_arr[1] = in_arr[2]
+    out_arr[2] = in_arr[0]
+    out_arr[3] = in_arr[5]
+    out_arr[4] = in_arr[3]
+    out_arr[5] = in_arr[4]
+    out_type = carray(out_dtype, 1)
+    out_type[0] = 6
```

Comment:
what is 6? Can we use one of the DALI types?

Reply:
We can't use DALI types in `nopython` mode function.